packages/dynamics-lib/src/entities/__tests__/pocl-validaton-error.entity.spec.js

```diff
@@ -0,0 +1,90 @@
+import { PoclValidationError } from '../pocl-validation-error.entity.js'
+import { retrieveGlobalOptionSets } from '../..'
+
+describe('pocl staging exception entity', () => {
+  let optionSetData
+  beforeAll(async () => {
+    optionSetData = await retrieveGlobalOptionSets().cached()
+  })
+  describe('maps from dynamics', () => {
+    let exception
+    beforeAll(() => {
+      exception = PoclValidationError.fromResponse(
+        {
+          '@odata.etag': 'W/"56351087"',
+          defra_poclvalidationerrorid: '91f15d18-0aa4-ea11-a812-000d3a64905b',
+          defra_firstname: 'Daniel',
+          defra_name: 'Ricciardo',
+          defra_organisation: 'Fishy Endeavours',
+          defra_premises: '14 Howecroft Court',
+          defra_street: 'Eastmead Lane',
+          defra_town: 'Bristol',
+          defra_postcode: 'BS9 1HJ',
+          defra_country: 'GB',
+          defra_birthdate: '1989-07-01',
+          defra_emailaddress: 'daniel-ricc@example.couk',
+          defra_mobilenumber: '07722 123456',
+          defra_preferredmethodofnewsletter: 910400003,
+          defra_preferredmethodofconfirmation: 910400000,
+          defra_preferredmethodofreminder: 910400002,
+          defra_concessions: '[{"type":"Blue Badge","referenceNumber":"123456789"}]',
+          defra_startdate: '2021-06-15',
+          defra_serialnumber: '14345-48457J',
+          defra_permitid: 'test-permit-id',
+          defra_transactiondate: '2020-01-01T14:00:00Z',
+          defra_amount: 30,
+          defra_paymentsource: 'Post Office Sales',
+          defra_channelid: '948594',
+          defra_methodofpayment: 910400001,
+          defra_status: 910400000,
+          defra_datasource: 910400000,
+          statecode: 1,
+          defra_errormessage: '\"permissions[0].licensee.email\" must be a valid email'
+        },
+        optionSetData
+      )
+    })
+
+    it('returns a PoclValidationError instance', () => {
+      expect(exception).toBeInstanceOf(PoclValidationError)
+    })
+
+    it('has the expected data', () => {
+      expect(exception).toMatchSnapshot()
+    })
```

Comment:
I've deliberated whether to make this comment - and it is just a comment, do with it what you will. I would do this slightly differently. Rather than having a `beforeAll` I'd have a `getSamplePoclValidationError` function, using this to get me an instance of the class, then assert on that. Something like this
```
it('returns a PoclValidationError instance', () => {
  const exception = getSampleValidationError()
  expect(exception).toBeInstanceOf(PoclValidationError)
})

it('has the expected data', () => {
  const exception = getSampleValidationError()
   expect(exception).toMatchSnapshot()
})
```
I say this because I'd have probably then used the `getSampleValidationError` method to do several different validation errors. 
However, I leave this as a comment, because I'm not all that sure that this would add a huge amount, or possibly even any, value

Reply:
I would normally use a `beforeEach` block and either have a function which creates the entity or create it directly in the `beforeEach`. I think I copied the structure of the test file from an existing file and forgot to change that bit.
I've changed the `beforeAll` to a `beforeEach` as I think that will give us the result we want with very little effort